src/module/loader.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) 2022-2023 SUSE LLC
+//
+// Author: Roy Hopkins <rhopkins@suse.de>
+
+extern crate alloc;
+
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use crate::error::SvsmError;
+use crate::fs::{list_dir, open, FileHandle};
+use crate::task::{create_elf_task, TaskPointer};
+
+pub struct ModuleLoader {
```

Comment:
Derive the usual suspects. Since it has a vector you probably need to manually implement Default. 
```rust
impl Default for ModuleLoader {
    fn default() -> Self {
        ModuleLoader {
            modules: Vec::new(),
        }
    }
}
```